service/src/main/java/bio/terra/landingzone/stairway/flight/create/resource/step/CreateSentinelStep.java

```diff
@@ -0,0 +1,72 @@
+package bio.terra.landingzone.stairway.flight.create.resource.step;
+
+import bio.terra.landingzone.library.landingzones.definition.ArmManagers;
+import bio.terra.landingzone.library.landingzones.definition.ResourceNameGenerator;
+import bio.terra.landingzone.library.landingzones.definition.factories.ParametersResolver;
+import bio.terra.landingzone.service.landingzone.azure.model.LandingZoneResource;
+import bio.terra.landingzone.stairway.flight.exception.MissingRequiredFieldsException;
+import bio.terra.stairway.FlightContext;
+import java.util.Optional;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class CreateSentinelStep extends BaseResourceCreateStep {
+  private static final Logger logger = LoggerFactory.getLogger(CreateSentinelStep.class);
+  public static final String SENTINEL_ID = "SENTINEL_ID";
+  public static final String SENTINEL_RESOURCE_KEY = "SENTINEL";
+
+  public CreateSentinelStep(
+      ArmManagers armManagers,
+      ParametersResolver parametersResolver,
+      ResourceNameGenerator resourceNameGenerator) {
+    super(armManagers, parametersResolver, resourceNameGenerator);
+  }
+
+  @Override
+  protected void createResource(FlightContext context, ArmManagers armManagers) {
+    var workspace =
+        getParameterOrThrow(
+            context.getWorkingMap(),
+            CreateLogAnalyticsWorkspaceStep.LOG_ANALYTICS_RESOURCE_KEY,
+            LandingZoneResource.class);
+
+    if (workspace.resourceName().isEmpty()) {
+      throw new MissingRequiredFieldsException("LogAnalyticsWorkspace resource name is not set.");
+    }
+
+    var state =
```

Comment:
Is there a resource provider that needs to be present for this, and if so can we check for it before the flight gets underway?

Reply:
@sergiygetlin adding that to the BPM list would require the provider for all landing zones, not just protected ones. I think we only want that for protected landing zones. 🤔 The more I think about it, it makes sense to pull the RP checks into the landing zone service out of BPM. The likely OOP would be:
1. Rawls receives BP creation request
2. Rawls fires a "validate MRG" request against the LZ service, which would verify things like quotas available and needed providers registered. 
3. If the check passes, proceed w/billing profile and then LZ creation
4. If the check fails, fail the request and give an error to the user. 

I think that'd be a separate ticket obviously, so probably fine as-is for now.

Reply:
Ticketed [here](https://broadworkbench.atlassian.net/browse/WOR-1008) for future reference.